UPMPackage/Scripts/Runtime/MixedTypographer.cs

```diff
@@ -0,0 +1,177 @@
+
+      for (int i = 0; i < isRtl.Length; i++) {
+        if (isRtl[i] == ContextType.Default) {
+          int ch = input.Get(i);
+
+          #region Judgment character condition
+
+          ContextType previousType = ContextType.Default;
+          ContextType behindType = ContextType.Default;
+          // White Space didn't affect text composing in current logic
+          // reserve for farther logic optimization
+          bool previousWhiteSpace = false;
+          bool behindWhiteSpace = false;
+
+          // search backward for Context.Type
+          // maybe a failed searching 
+          for (int j = 1; j <= i; j++) {
+            if (isRtl[i - j] != ContextType.Default) {
+              previousType = isRtl[i - j];
+
+              break;
+            } else if (input.Get(i - j) == ' ') {
+              previousWhiteSpace = true;
+            }
+          }
+
+          // search forward for Context.Type
+          // maybe a failed searching
+          for (int j = 1; j + i <= input.Length - 1; j++) {
+            if (isRtl[i + j] != ContextType.Default) {
+              behindType = isRtl[i + j];
+              break;
+            } else if (input.Get(i + j) == ' ') {
+              behindWhiteSpace = true;
+            }
+          }
+
+          if (previousType == ContextType.Default && behindType != ContextType.Default) {
+            previousType = behindType;
+          } else if (previousType != ContextType.Default && behindType == ContextType.Default) {
+            behindType = previousType;
+          }
+
+          #endregion
+
+          #region Mark Mirrored Character
+
+          if (MirroredCharsMaper.MirroredCharsMap.ContainsKey((char)ch)) {
+            var mirrorCharacter = MirroredCharsMaper.MirroredCharsMap[(char)ch];
+            if (mirrorCharacter > (char)ch) {
+              for (int j = 1; j < input.Length - i; j++) {
+                if (input.Get(i + j) == mirrorCharacter) {
+                  ContextType mirrorPreviousType = ContextType.Default;
+                  ContextType mirrorBehindType = ContextType.Default;
+                  for (int k = 1; k <= j - 1; k++) {
+                    if (isRtl[i + j - k] != ContextType.Default) {
+                      mirrorPreviousType = isRtl[i + j - k];
+                      break;
+                    }
+                  }
+
+                  for (int k = 1; k + i + j < input.Length; k++) {
+                    if (isRtl[i + j + k] != ContextType.Default) {
+                      mirrorBehindType = isRtl[i + j + k];
+                      break;
+                    }
+                  }
+
+                  //if right character is rightest, case previous type only
+                  if (i + j == input.Length - 1) behindType = mirrorPreviousType;
+                  //if previous type is default, there is no letter in or front this mirror character
+                  if (mirrorPreviousType == ContextType.Default) mirrorPreviousType = mirrorBehindType;
```

Comment:
```suggestion
                  if (mirrorPreviousType == ContextType.Default) {
                    mirrorPreviousType = mirrorBehindType;
                  }
```
A maximum of one statement per line. According to [C# at Google Style Guide](https://google.github.io/styleguide/csharp-style.html)